### PR TITLE
pkg/ocm: remove AWS client from create cluster call

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Every caller of ocm.CreateCluster has an AWS client already, and loading one implicitly in the call makes it difficult for the ocm library to be re-used, especially in environments where the common AWS SDK credentials may not live in the user-focused locations.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@ciaranRoche happy to go either way - just let me know what you'd prefer!

**Comment by hunterkepley** [comment]
LGTM- I assume codecov is failing due to low coverage. Overall changes look fine to me

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -3141,6 +3142,23 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	if clusterConfig.CustomProperties != nil && clusterConfig.CustomProperties[properties.UseLocalCredentials] == "true" {
```
Comment (gdbranco) [inline cmd/create/cluster/cmd.go:3145]:
```suggestion
	if clusterConfig.CustomProperties != nil && clusterConfig.CustomProperties[properties.UseLocalCredentials] == strconv.FormatBool(true) {
```

**Comment by ciaranRoche** [comment]
@stevekuznetsov changes look good, if we could get some tests, we are good to merge then

File: pkg/ocm/clusters.go [hunk 2]
```diff
@@ -114,6 +113,10 @@ type Spec struct {
 	// Disable SCP checks in the installer by setting credentials mode as mint
 	DisableSCPChecks *bool
 
+	// Non-STS
+	AWSAccessKey *aws.AccessKey
+	AWSCreator   *aws.Creator
```
Comment (stevekuznetsov) [inline pkg/ocm/clusters.go:118]: @ciaranRoche one question - the other `pkg/ocm` client calls take this in as a parameter - I know you mentioned putting in the `struct` here, but wanted to check that there wasn't a preference one way or the other

File: cmd/create/cluster/spec_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+func (f fakeGetter) GetLocalAWSAccessKeys() (*aws.AccessKey, error) {
+	if f.localAccessKeys != nil {
+		return f.localAccessKeys, nil
+	}
+	return nil, fmt.Errorf("no AWS access keys configured on fake")
+}
+
+var _ aws.AccessKeyGetter = (*fakeGetter)(nil)
+
+func TestClusterConfigFor(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply cmd/create/cluster/spec_test.go]: Done!

**Comment by stevekuznetsov** [comment]
@robpblake updated for Ginkgo+Gomega

**Review by robpblake — APPROVED** [review]
/lgtm

**Review by ciaranRoche — APPROVED** [review]
/lgtm

**Pull request merged**